blocks/api/raw-handling/is-inline-content.js

```diff
@@ -1,7 +1,56 @@
+/**
+ * External dependencies
+ */
+import { includes } from 'lodash';
+
 /**
  * Internal dependencies
  */
-import { isInline, isDoubleBR } from './utils';
+import { isPhrasingContent } from './utils';
+
+/**
+ * An array of tag groups used by isInlineForTag function.
+ * If tagName and nodeName are present in the same group, the node should be treated as inline.
+ * @type {Array}
+ */
+const inlineWhitelistTagGroups = [
+	[ 'ul', 'li', 'ol' ],
+	[ 'h1', 'h2', 'h3', 'h4', 'h5', 'h6' ],
+];
+
+/**
+ * Checks if nodeName should be treated as inline when being added to tagName.
+ * This happens if nodeName and tagName are in the same group defined in phrasingContentTagGroups.
+ *
+ * @param {string} nodeName Node name.
+ * @param {string} tagName  Tag name.
+ *
+ * @return {boolean} True if nodeName is inline in the context of tagName and
+ *                    false otherwise.
+ */
+function isInlineForTag( nodeName, tagName ) {
+	if ( ! tagName || ! nodeName ) {
+		return false;
+	}
+	return inlineWhitelistTagGroups.some( ( tagGroup ) =>
+		includes( tagGroup, nodeName ) && includes( tagGroup, tagName )
```

Comment:
Minor: Micro-optimization: There _might_ be a small impact in avoiding redundant iteration by using [`_.difference`](https://lodash.com/docs/4.17.5#difference):

```js
! difference( [ tagName, nodeName ], tagGroup ).length
```

But also likely harder to interpret at a glance.

Reply:
Interesting!